docs/src/main/paradox/hdfs.md

```diff
@@ -0,0 +1,234 @@
+# Hadoop Distributed File System - HDFS
+
+The connector offers Flows and Sources that interact with HDFS file systems.
+
+For more information about Hadoop, please visit the [Hadoop documentation](https://hadoop.apache.org/).
+
+### Reported issues
+
+[Tagged issues at Github](https://github.com/akka/alpakka/labels/p%3Ahdfs)
+
+## Artifacts
+
+@@dependency [sbt,Maven,Gradle] {
+  group=com.lightbend.akka
+  artifact=akka-stream-alpakka-hdfs_$scalaBinaryVersion$
+  version=$version$
+}
+
+## Specifying a Hadoop Version
+
+By default, HDFS connector uses Hadoop **3.1.0**. If you are using a different version of Hadoop, 
+you should exclude the Hadoop libraries from the connector dependency and add the dependency for your preferred version.
+
+## Set up client
+
+Flows provided by this connector need a prepared `org.apache.hadoop.fs.FileSystem` to 
+interact with HDFS.
+
+
+Scala
+: @@snip ($alpakka$/hdfs/src/test/scala/akka/stream/alpakka/hdfs/HdfsWriterSpec.scala) { #init-client }
+
+Java
+: @@snip ($alpakka$/hdfs/src/test/java/akka/stream/alpakka/hdfs/HdfsWriterTest.java) { #init-client }
+
+
+## Writing
+
+The connector provides three Flows. Each flow requires `RotationStrategy` and `SyncStrategy` to run.
+@scala[@scaladoc[HdfsFlow](akka.stream.alpakka.hdfs.scaladsl.HdfsFlow$).]
+@java[@scaladoc[HdfsFlow](akka.stream.alpakka.hdfs.javadsl.HdfsFlow$).]
+
+The flows push `OutgoingMessage` to a downstream.
+
+### Data Writer
+
+Use `HdfsFlow.data` to stream with `FSDataOutputStream` without any compression.
+
+
+Scala
+: @@snip ($alpakka$/hdfs/src/test/scala/akka/stream/alpakka/hdfs/HdfsWriterSpec.scala) { #define-data }
+
+Java
+: @@snip ($alpakka$/hdfs/src/test/java/akka/stream/alpakka/hdfs/HdfsWriterTest.java) { #define-data }
+
+
+### Compressed Data Writer
+
+First, create `CompressionCodec`.
```

Comment:
By adding 
```
"javadoc.org.apache.hadoop.base_url" -> s"https://hadoop.apache.org/docs/r${hadoopVersion}/api/",
```
in `build.sbt` you can create links to Hadoop's API via `@javadoc`.